File: ruby/word-count/word_count.rb
# All the phrases and all
class Phrase
  attr_reader :phrase

  def initialize(phrase)
    @phrase = phrase
  end

  def word_count
    phrase.strip.split(/[\s,]+/).each_with_object({}) do |word, object|
      key = word.gsub(/\A[\W]+|[\W]+\z/, '').downcase
      object[key] = object[key].to_i + 1
    end
  end
end

File: typescript/word-count/word-count.ts
export default class Word {
  count(phrase: string): Map<string, number> {
    const wordCount: Map<string, number> = new Map()

    phrase
      .toLowerCase()
      .trim()
      .split(/\s+/)
      .forEach((curr) => {
        const count: number | undefined = wordCount.get(curr)
        count ? wordCount.set(curr, count + 1) : wordCount.set(curr, 1)
      })

    return wordCount
  }
}